src/App.tsx

```diff
@@ -1,34 +1,108 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Filter } from './types/Filter';
+
+function getVisibleTodos(todos: Todo[], query: string, filter: string) {
+  let preparedTodos = [...todos];
+  const normalizedQuery = query.toLowerCase().trim();
```

Comment:
in an example of a working page our array changes if you type spaces. so, I'm not sure, if we should use trim().